### DISCUSSION TITLE
Executing jobs interactively, rather than as batch jobs.

### DISCUSSION THREAD

**Discussion by cameronsmith1** [E3SM-Project/zppy#296] [Show and tell]
Running interactively, rather than using the queue system, can be desirable.   For example, on some machines, eg Cori, the queue system cause significant wait times.  Running interactively may also help with debugging.

Here is what I have found useful:

In the configuration file, set the following to create the bash files without submitting them to the queue:

dry_run = true

In the directory with all of the bash scripts, the scripts need to be edited to remove variables that would normally be set by the queue system.  I have found the following shell commands work for me:

sed -i -s -e '/id=${SLURM_JOBID}/c id=$RANDOM' *.bash
sed -i -s -e '/--jobs=${SLURM_NNODES}/d' *.bash

When running interactively, it is often desirable to using a single processor, so I found the following helpful.

sed -i -s -e '/--parallel=mpi/c --parallel=serial \\' *.bash

NOTE: Some bash scripts need to be run before others, but I found it easy to figure out the order with a little trial and error.

**Comment by forsyth2** [comment]
#388, once implemented, should also help with this.
